qiskit_optimization/translators/ising.py

```diff
@@ -0,0 +1,255 @@
+    sense = quad_prog.objective.sense.value
+
+    # convert a constant part of the object function into Hamiltonian.
+    offset += quad_prog.objective.constant * sense
+
+    # convert linear parts of the object function into Hamiltonian.
+    for idx, coef in quad_prog.objective.linear.to_dict().items():
+        z_p = zeros(num_nodes, dtype=bool)
+        weight = coef * sense / 2
+        z_p[idx] = True
+
+        pauli_list.append([-weight, Pauli((z_p, zero))])
+        offset += weight
+
+    # convert quadratic parts of the object function into Hamiltonian.
+    # first merge coefficients (i, j) and (j, i)
+    coeffs = {}  # type: Dict
+    for (i, j), coeff in quad_prog.objective.quadratic.to_dict().items():
+        if j < i:  # type: ignore
+            coeffs[(j, i)] = coeffs.get((j, i), 0.0) + coeff
+        else:
+            coeffs[(i, j)] = coeffs.get((i, j), 0.0) + coeff
+
+    # create Pauli terms
+    for (i, j), coeff in coeffs.items():
+
+        weight = coeff * sense / 4
+
+        if i == j:
+            offset += weight
+        else:
+            z_p = zeros(num_nodes, dtype=bool)
+            z_p[i] = True
+            z_p[j] = True
+            pauli_list.append([weight, Pauli((z_p, zero))])
+
+        z_p = zeros(num_nodes, dtype=bool)
+        z_p[i] = True
+        pauli_list.append([-weight, Pauli((z_p, zero))])
+
+        z_p = zeros(num_nodes, dtype=bool)
+        z_p[j] = True
+        pauli_list.append([-weight, Pauli((z_p, zero))])
+
+        offset += weight
+
+    # Remove paulis whose coefficients are zeros.
+    qubit_op = sum(PauliOp(pauli, coeff=coeff) for coeff, pauli in pauli_list)
+
+    # qubit_op could be the integer 0, in this case return an identity operator of
+    # appropriate size
+    if isinstance(qubit_op, OperatorBase):
+        qubit_op = qubit_op.reduce()
+    else:
+        qubit_op = I ^ num_nodes
+
+    return qubit_op, offset
+
+
+def from_ising(
+    qubit_op: Union[OperatorBase, PauliSumOp],
+    offset: float = 0.0,
+    linear: bool = False,
+) -> QuadraticProgram:
+    r"""Create a quadratic program from a qubit operator and a shift value.
+
+    Variables are mapped to qubits in the same order, i.e.,
+    i-th variable is mapped to i-th qubit.
+    See https://github.com/Qiskit/qiskit-terra/issues/1148 for details.
+
+    Args:
+        qubit_op: The qubit operator of the problem.
+        offset: The constant value in the Ising Hamiltonian.
+        linear: If linear is True, :math:`x^2` is treated as a linear term
+            since :math:`x^2 = x` for :math:`x \in \{0,1\}`.
+            Else, :math:`x^2` is treat as a quadratic term.
+            The default value is False.
+
+    Returns:
+        The quadratic program corresponding to the qubit operator.
+
+    Raises:
+        QiskitOptimizationError: If there are Pauli Xs in any Pauli term
+        QiskitOptimizationError: If there are more than 2 Pauli Zs in any Pauli term
```

Comment:
Here its using capital X and Z. In the code its checking Pauli.x and Pauli.z the .x will capture any X or Ys. I am just wondering about the text here - should it say 'If there are Pauli Xs or Ys in any term' which seems more correct to me

Reply:
I agree. I put your expression "If there are Pauli Xs or Ys in any term".